src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java

```diff
@@ -226,8 +229,8 @@ public void testNonExistingOutputFormat() throws Exception {
     public void testNonExistingClass() throws Exception {
         exit.expectSystemExitWithStatus(-2);
         exit.checkAssertionAfterwards(() -> {
-            final String expectedExceptionMessage =
-                    String.format(Locale.ROOT, "Checkstyle ends with 1 errors.%n");
+            final String expectedExceptionMessage = errorCounterMessage.getMessage()
+                    + System.getProperty("line.separator");
```

Comment:
`System.getProperty("line.separator")` is used too much , please move it to field with name `EOL`, substitute in all cases